contracts/genie-raffle/src/contract.rs

```diff
@@ -0,0 +1,391 @@
+    ]))
+}
+
+pub fn handle_increase_native_incentives(
+    deps: DepsMut,
+    env: Env,
+    info: MessageInfo,
+) -> Result<Response, StdError> {
+    if query_status(deps.as_ref(), &env)?.status != Status::NotStarted {
+        return Err(StdError::generic_err(
+            "rewards can only be deposited before campaign starts",
+        ));
+    }
+
+    let config = CONFIG.load(deps.storage)?;
+    if info.sender != config.owner {
+        return Err(StdError::generic_err("can only be called by owner"));
+    }
+
+    let increase_amount: Uint128 = match config.asset.clone() {
+        AssetInfo::NativeToken { denom } => info
+            .funds
+            .iter()
+            .filter(|coin| coin.denom == denom)
+            .map(|coin| coin.amount)
+            .sum(),
+        AssetInfo::Token { contract_addr: _ } => {
+            return Err(StdError::generic_err("invalid asset type"));
+        }
+    };
+    if increase_amount.is_zero() {
+        return Err(StdError::generic_err("amount must be greater than 0"));
+    }
+
+    let mut state = STATE.load(deps.storage)?;
+    state.protocol_funding += increase_amount;
+    STATE.save(deps.storage, &state)?;
+
+    Ok(Response::new().add_attributes(vec![
+        attr("action", "genie_increase_rewards"),
+        attr("asset", config.asset.asset_string()),
+        attr("protocol_funding", state.protocol_funding),
+    ]))
+}
+
+pub fn handle_claim(
+    deps: DepsMut,
+    env: Env,
+    info: MessageInfo,
+    claim_amounts: Binary,
+    signature: Binary,
+) -> Result<Response, StdError> {
+    if query_status(deps.as_ref(), &env)?.status != Status::Ongoing {
+        return Err(StdError::generic_err("campaign is not ongoing"));
+    }
+
+    // convert claim_amounts to string
+    let claim_string = String::from_utf8(claim_amounts.to_vec())?;
+    let claim_string_array = claim_string.split(",").collect::<Vec<&str>>();
+    let claim_amounts = claim_string_array
+        .iter()
+        .map(|x| x.parse::<Uint128>())
+        .collect::<Result<Vec<Uint128>, _>>()?;
+
+    let recipient = &info.sender;
+    let mut user_info = USERS.load(deps.storage, recipient).unwrap_or_default();
+    let mut state = STATE.load(deps.storage)?;
+
+    if claim_amounts.len() != state.unclaimed_amounts.len() {
+        return Err(StdError::generic_err(
+            "claim amount length does not match claimable amount length",
+        ));
+    }
+
+    // Check if signature is valid
+    let is_valid = is_valid_signature(
+        &deps,
+        &recipient,
+        &env.contract.address.to_string(),
+        claim_amounts.clone(),
+        &signature,
+        &CONFIG.load(deps.storage)?.public_key,
+    )?;
+    if !is_valid {
+        return Err(StdError::generic_err("signature verification failed"));
+    }
+
+    let mut claimable_amounts: Vec<Uint128> = vec![];
+    // iterate through claimed_amounts and claim_amount to verify that claim_amount is greater than claimed_amount
```

Comment:
```suggestion
    // iterate through claimed_amounts and claim_amount to verify that claim_amount is greater than/equal claimed_amount
```
same question as earlier comment